app/lib/payment_gateways/stripe_crypt.rb

```diff
@@ -38,13 +36,9 @@ def create_stripe_setup_intent
 
     delegate :payment_detail, to: :account
 
-    def customer_id
-      payment_detail.credit_card_auth_code
-    end
-
     def find_or_create_customer
-      customer = Stripe::Customer.retrieve(customer_id, api_key) if customer_id
-      customer.try(:id) ? customer : create_customer
+      customer_id = payment_detail.credit_card_auth_code
+      customer_id.present? ? Stripe::Customer.retrieve(customer_id, api_key) : create_customer
```

Comment:
And what happens when Stripe does not find the customer? According to https://github.com/stripe/stripe-ruby/blob/9787913b355049af915cf075b29666e55c5ab7f2/lib/stripe/api_resource.rb#L103, it will _build_ one. The idea here before was to ensure the customer exists in Stripe and therefore `customer.try(:id)`.

Reply:
Let's move this conversation to https://github.com/3scale/porta/pull/2342#discussion_r563567816